src/Service/Client/ServiceClient.php

```diff
@@ -94,8 +110,11 @@ public function request(
     /**
      * @throws LtiExceptionInterface
      */
-    private function getAccessToken(RegistrationInterface $registration, array $scopes): string
-    {
+    private function getAccessToken(
+        RegistrationInterface $registration,
+        array $scopes,
+        bool $forceRefresh = false
```

Comment:
Consider adding an explicit method to delete a cached access key instead of using a behavior-modifying flag.
Right now the method is violating SRP by both modifying a state _and_ returning a value at the same time.

Reply:
it's a private method that already changfe the state and return a value (it caches when it gets a token).

if you prefer I can trigger the cache deletion in the main method just before recalling the access token endpoint, but make the code a bit less clear imo.

Reply:
All right, considering that caching-logic grouping under the method, the approach is indeed justified. Sorry for getting all nitpicky here.

Reply:
I have absolutely no problems with relevant remarks :)